### ISSUE TITLE
Handlebars Engine compilation of handlebars to javascript failed

### ISSUE THREAD

**Issue by tcrossland** [bertramdev/asset-pipeline#215]
After upgrading a grails 2.4.4 application to asset-pipeline 2.1.0 and handlebars-asset-pipeline 1.3.0.5, the following error occurs during the asset compilation phase of "grails war":

```
...
|Processing File 18 of 44 - application.js
.Error
|
WAR packaging error:
                        Handlebars Engine compilation of handlebars to javascript failed.
                        java.lang.NullPointerException: Cannot invoke method split() on null object
```

I've created the following example app which reproduces the issue:

https://github.com/tcrossland/handlebars-asset-pipeline-example

**Comment by davydotcom** [comment]
The issue is a bug in handlebars-asset-pipeline due to the placement of your template files at the root. Due to the fact that there is no parentPath. This is remedied in handlebars-asset-pipeline:2.0.0 (grails version). I will publish it as soon as maven central propagates

**Comment by davydotcom** [comment]
Released. feel free to let me know if it isn't fixed

**Comment by tcrossland** [comment]
Unfortunately it's still failing, but the message has changed:

```
|Processing File 18 of 44 - application.js
.Error
|
WAR packaging error:
                        Handlebars Engine compilation of handlebars to javascript failed.
                        org.mozilla.javascript.EcmaError: ReferenceError: "Handlebars" is not defined.
```

I've updated the example application.

**Comment by davydotcom** [comment]
hmm 2.0 of handlebars not able to find handlebars.js ? ill check real fast

**Comment by tcrossland** [comment]
Hmmm... I've taken a quick look and it appears that the handelbars.js which is being loaded into the Rhino context in HandlebarsProcessor is the one that says:

```
//= require handlebars/handlebars.js
```

The handlebars/handlebars.js doesn't seem to get loaded into the context, thus Handlebars remains undefined.

**Comment by tcrossland** [comment]
As a hacky workaround, I've set the `scanPath` configuration option to some meaningless value:

```
grails.assets.handlebars.scanPath='xxx'
```

`HandleBarsProcessor` now loads the correct `handlebars.js`.

**Comment by davydotcom** [comment]
Ah I meant to move that

**Comment by davydotcom** [comment]
Hehe just saw your workaround and well played. Yep the idea is the new handlebars won't be coupled to a specific version. It can use whatever version you have

**Comment by tcrossland** [comment]
I guessed that was the idea (good idea btw), but was too tired to figure out how to use scanPath correctly. Mañana más...

**Comment by tcrossland** [comment]
Just to confirm, this is now working with handlebars-grails-asset-pipeline 2.0.0.1, with the fix in handlebars-asset-pipeline 2.1.1. FYI, the 2.1.1 sources seem to be on maven central but not in the github repo.

Thanks for the fix!

**Comment by davydotcom** [comment]
Np, Thanks for the heads up I had to fight bintray for a day. There were some issues with the account.
--
David Estes

On January 24, 2015 at 9:15:45 AM, Tom Crossland (notifications@github.com) wrote:

Just to confirm, this is now working with handlebars-grails-asset-pipeline 2.0.0.1, with the fix in handlebars-asset-pipeline 2.1.1. FYI, the 2.1.1 sources seem to be on maven central but not in the github repo.

Thanks for the fix!
